File: techila/examples/python/Features/basic_monte_carlo_pi/run_mcpi.py
# Copyright 2012-2016 Techila Technologies Ltd.

# This Python script contains the Local Control Code, which will be
# used to distribute computations to the Techila environment.
#
# The Python script named 'mcpi_dist.py' will be distributed and
# evaluated on Workers. The 'loops' parameter will be transferred to all
# Jobs as an input argument. The 'run_mcpi' function will return the value
# of the 'result' variable, which will contain the approximated value of Pi.
#
# Usage:
# execfile('run_mcpi.py')
# result = run_mcpi(jobcount, loops)
# jobcount: number of Jobs in the Project
# loops: number of iterations performed in each Job
#
# Example:
# result = run_mcpi(10, 100000)

# Load the techila package
import techila

def run_mcpi(jobcount, loops):

    # Create the computational Project with the peach function.
    result = techila.peach(funcname = 'mcpi_dist', # Function that will be executed on Workers
                           params = [loops], # Parameters for the executable function
                           files = ['mcpi_dist.py'], # Files that will be evaluated on Workers
                           jobs = jobcount # Number of Jobs in the Project
                           )

    # Calculate the approximated value of Pi based on the results received.
    result = 4 * float(sum(result)) / (jobcount * loops)

    # Display results
    print('The approximated value of Pi is:', result)
    return(result)

File: techila/examples/R/Features/basic_monte_carlo_pi/run_mcpi.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This R-script contains the Local Control Code, which will be used to
# distribute computations to the Techila environment.
#
# The R-script named "mcpi_dist.r" will be distributed and sourced on
# Workers. The 'loops' parameter will be transferred to all Jobs. The
# peachvector will be used to control the number of Jobs in the
# Project. The 'run_mcpi' function will return the value of the
# 'result' variable, which will contain the approximated value of Pi.
#
# Usage:
# source("run_mcpi.r")
# result <- run_mcpi(jobs, loops)
# jobs: number of Jobs in the Project
# loops: number of Monte Carlo approximations performed per Job
#
# Example:
# result <- run_mcpi(10, 100000)

# Load the techila library
library(techila)

run_mcpi <- function(jobs, loops) {

  # Create the computational Project with the peach function.
  result <- peach(funcname = "mcpi_dist", # Function that will be executed on Workers
                  params = list(loops), # Parameters for the executable function
                  files = list("mcpi_dist.r"), # Files that will be sourced on Workers
                  peachvector = 1:jobs, # Length of the peachvector determines the number of Jobs.
                  sdkroot = "../../../..") # Location of the techila_settings.ini file

  # Calculate the approximated value of Pi based on the generated data.
  result <- 4 * sum(as.numeric(result)) / (jobs * loops)

  # Display results
  print(c("The approximated value of Pi is:", result))
  result
}
